src/plugins/intel_gpu/src/plugin/program_builder.cpp

```diff
@@ -304,6 +304,16 @@ void ProgramBuilder::add_primitive(const ov::Node& op, std::shared_ptr<cldnn::pr
     prim->origin_op_name = op.get_friendly_name();
     prim->origin_op_type_name = op.get_type_name();
 
+    if (auto data_prim = dynamic_cast<cldnn::data*>(prim.get())) {
+        auto rt_info = op.get_rt_info();
+        auto offset = rt_info.find("bin_offset");
```

Comment:
I ensure it partially in two ways: 	
	1. I save bin_offset in Node's rt_info before the transformation pipeline is launched. This way if a Node is replaced by a new one during the trasformation pipeline, its values are cached in the traditional way. 
	2. During saving to cache I check if the byte size of the data has changed since the model was loaded - this mostly targets constants affected by the precision transformations.
		
I understand that this is a limited approach and there is a potential for more situations akin to the one with strided_slice in topologies I have not tested. I asked about it in the email thread you were just added to and received a response that it looks satisfactory. I you think it's not - do you have any suggestions on how to achieve it? I couldn't think of a 100% foolproof solution that does not involve direct comparison of the data (which would defeat the purpose).

Reply:
@vladimir-paramuzov I replaced the strided slice WA with a solution that universally prevents weightless caching for constants which are replaced during the transformation pipeline, even if the rt_info is explicitly copied from the old one to the new. Can you think of any other scenarios where this is not sufficient? Specifically, is it possible for constant's data to be altered without replacing a node with a new one during the transformation pipeline?